### ISSUE TITLE
Opening/closing tags don't match using vscode's bracketPairColorization

### ISSUE THREAD

**Issue by bjarkihall** [microsoft/AL#6755]
Please include the following with each issue:

**1. Describe the bug**
Brackets don't match using vscode's bracket pair colorization.

**2. To Reproduce**
Set "editor.bracketPairColorization.enabled": true in vscode settings (you need to have at least version 1.60).

You can try different kinds of blocks (and I provided a screenshot of the results):

```al
codeunit 50000 FooCodeunit
{
    var
        _Foo: Integer;

    procedure Bar()
    begin
        Message('??');
        if _Foo = 1 then begin
            Message('???');
        end;
    end;

    procedure IsPending()
    begin
        Message('????');
    end;

    procedure BeginProcedure()
    begin
        case true of
            1 = 1:
                Message('?????');
        end
    end;
}
```

**3. Expected behavior**
I expect the opening/closing colors to match, currently it's very easy to break.

**4. Actual behavior**
Only actual "brackets" (e.g. object and field definitions) and parentheses (for function declarations/invocations) seem to open/close correctly.

![image](https://user-images.githubusercontent.com/8615374/131928018-70021ba0-e5f3-4384-bcdc-9f32075ba6f5.png)

**5. Versions:**

- AL Language:
v8.0.508407
- Visual Studio Code:
1.60
- Business Central:
18.3
- List of Visual Studio Code extensions that you have installed:
AL Language v8.0.508407

### Final Checklist

Please remember to do the following:

* [x] Search the issue repository to ensure you are reporting a new issue

* [x] Reproduce the issue after disabling all extensions except the AL Language extension

* [x] Simplify your code around the issue to better isolate the problem

**Comment by bjarkihall** [comment]
Please note, that I'm not using the Bracket-Colorizer extension, this came built-in with the latest vscode update:
https://code.visualstudio.com/updates/v1_60#_high-performance-bracket-pair-colorization

**Comment by bjarkihall** [comment]
Seems like other languages are experiencing these kind of problems for this release, I'm not sure if the AL extension needs new definitions for colorizing to work properly  but at least I'll keep the ticket open as a heads up. For reference: https://github.com/microsoft/vscode/issues/132164

**Comment by nndobrev** [comment]
Hi @bjarkihall, I installed the latest VSC and tested but unfortunately I don't have a repro. 

![image](https://user-images.githubusercontent.com/85873063/132015946-67e2b928-154d-4cc0-b6ee-4c87658cf302.png)


Are you sure that you only have the AL Language extension installed?

**Comment by salgiza** [comment]
Hi @nndobrev, as @bjarkihall mentioned in the ticket, you have to set editor.bracketPairColorization.enabled to true in VS Code settings. Currently it's not enabled by default.

**Comment by kalberes** [comment]
Correct. We simply do not support this new option (yet).

**Comment by bjarkihall** [comment]
@kalberes yes, the intention of the ticket was to act as a heads-up, since devs might read the release notes for vscode 1.60 and try this new feature out (it might become a default later on, who knows?).

I think the AL Language extension could add the new `colorizedBracketPairs` property to AL's language definition, but you don't need to rush it until the API is properly documented (which is isn't yet it seems): https://code.visualstudio.com/api/language-extensions/language-configuration-guide

These will possibly become user-defined in the next vscode release: https://github.com/microsoft/vscode/issues/131412 which would enable a work-around until this becomes supported by the AL Language extension itself.

**Comment by SBalslev** [comment]
Hi! Thank you for reporting this issue with VS Code's bracket pair colorization and AL syntax — we appreciate the clear description.

We can confirm this was addressed internally (PR 143438) by improving syntax highlighting to not highlight brackets within quoted identifiers, shipped in **Business Central runtime 22.0 (2023 Wave 1)**.

We're closing this as resolved. If bracket pair colorization still behaves incorrectly on the latest AL Language extension, please open a new issue.

Thank you for helping us improve the AL developer experience!
